server/webapp/WEB-INF/rails/webpack/views/components/environment_variables/index.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {MithrilComponent, MithrilViewComponent} from "jsx/mithril-component";
+import m from "mithril";
+import {EnvironmentVariable, EnvironmentVariables} from "models/environment_variables/types";
+import {ButtonIcon, Secondary} from "views/components/buttons";
+import {  SimplePasswordField, TextField} from "views/components/forms/input_fields";
+import {Close} from "views/components/icons";
+import styles from "./index.scss";
+
+interface EnvironmentVariableWidgetAttrs {
+  environmentVariable: EnvironmentVariable;
+  onRemove: (environmentVariable: EnvironmentVariable) => void;
+}
+
+export class EnvironmentVariableWidget extends MithrilViewComponent<EnvironmentVariableWidgetAttrs> {
+  view(vnode: m.Vnode<EnvironmentVariableWidgetAttrs>): m.Children {
+    let valueField: m.Children;
+
+    if (vnode.attrs.environmentVariable.secure()) {
+      valueField = <SimplePasswordField property={vnode.attrs.environmentVariable.encryptedValue}
```

Comment:
This seems wrong as we are calling `EnvironmentVariableWidget` with either `plainTextVaraible` or the `secureVariable`. There should be better way of doing this.